src/thor/triplegbuilder.cc

```diff
@@ -481,6 +481,40 @@ void AddTransitNodes(TripLeg_Node* trip_node,
   }
 }
 
+void SetTripEdgeRoadClass(TripLeg_Edge* trip_edge,
+                          const DirectedEdge* directededge,
+                          const GraphTile* graphtile,
+                          GraphReader& graphreader) {
+  trip_edge->set_road_class(GetTripLegRoadClass(directededge->classification()));
+  // If this is a ramp it may have been reclassified in graph enhancer.
+  // To restore the original road class for motorway_links, we check if any of the adjacent edges is
+  // a motorway.
+  if (directededge->use() == Use::kRamp) {
+    auto edge_nodes = graphreader.GetDirectedEdgeNodes(graphtile, directededge);
+    for (const auto& edge_node : {edge_nodes.first, edge_nodes.second}) {
```

Comment:
need a chekc at the top of the loop whether or not the edge_node is valid, if not continue

Reply:
also dont need const & because its a 64bit integer